chain-storage/src/block_info.rs

```diff
@@ -0,0 +1,95 @@
+use std::io::{Read, Write};
+
+#[derive(Clone)]
+pub struct BlockInfo {
+    id: Box<[u8]>,
+    parent_id: Box<[u8]>,
+    chain_length: u32,
+    ref_count: u32,
+}
+
+impl BlockInfo {
+    pub fn new(id: Vec<u8>, parent_id: Vec<u8>, chain_length: u32) -> Self {
```

Comment:
The `Vec` parameters could be generalized to something like `impl Into<Box<[u8]>>`.